packages/report-flaky-tests/lib/run.ts

```diff
@@ -1,148 +1,116 @@
 		artifactNamePrefix
 	);
 
 	if ( ! flakyTests ) {
+		// No flaky tests reported in this run.
 		return;
 	}
 
-	const issues = await fetchAllIssuesLabeledFlaky( octokit, label );
+	const label = core.getInput( 'label', { required: true } );
+	const issues = await api.fetchAllIssuesLabeledFlaky( label );
 
 	for ( const flakyTest of flakyTests ) {
-		const {
-			runner: testRunner = 'jest-circus',
-			title: testTitle,
-			path: testPath,
-			results: testResults,
-		} = flakyTest;
+		const { title: testTitle } = flakyTest;
 		const issueTitle = getIssueTitle( testTitle );
 		const reportedIssue = issues.find(
 			( issue ) => issue.title === issueTitle
 		);
-		const isTrunk = getHeadBranch() === 'trunk';
+		const testPath = flakyTest.path.startsWith( process.cwd() )
+			? flakyTest.path.slice( process.cwd().length )
+			: flakyTest.path;
 		let issue;
 
-		if ( reportedIssue ) {
-			const body = reportedIssue.body;
-			const meta = metaData.get( body );
+		const currentFormattedTestResults = formatTestResults( {
+			date: new Date(),
+			failedTimes: flakyTest.results.length,
+			headBranch: api.headBranch,
+			runURL: api.runURL,
+			// Always output the latest test results' stacks.
+			errorMessage: formatTestErrorMessage( flakyTest ),
+		} );
 
-			if ( isTrunk ) {
-				const headCommit = github.context.sha;
-				const baseCommit = meta.baseCommit || github.context.sha;
-				meta.baseCommit = baseCommit;
+		if ( reportedIssue ) {
+			const body = reportedIssue.body!;
 
+			// The issue is closed.
+			if ( reportedIssue.closed_at ) {
 				try {
-					const { data } = await octokit.rest.repos.compareCommits( {
-						...github.context.repo,
-						base: baseCommit,
-						head: headCommit,
-						per_page: 1,
-					} );
-
-					meta.failedTimes += testResults.length;
-					meta.totalCommits = data.total_commits + 1;
+					// Represent the latest base commit on trunk.
+					const latestAncestorCommit = await api.findMergeBaseCommit(
+						'trunk',
+						github.context.sha
+					);
+					const latestAncestorDate =
+						latestAncestorCommit.committer?.date;
+
+					// Cannot find the commit date, skipping.
+					if ( ! latestAncestorDate ) {
+						return;
+					}
+
+					const closedTime = new Date(
+						reportedIssue.closed_at
+					).getTime();
+					const latestAncestorTime = new Date(
+						latestAncestorDate
+					).getTime();
+
+					// The issue is closed after the latest base commit on trunk,
+					// which means the branch/PR/commit is outdated and the flaky test
+					// has probably already been fixed. Skip reporting the outdated flaky tests.
+					if ( closedTime >= latestAncestorTime ) {
+						return;
+					}
 				} catch ( err ) {
-					// It might be a deleted commit,
-					// treat the current commit as the base commit.
-					meta.baseCommit = headCommit;
-					meta.failedTimes = testResults.length;
-					meta.totalCommits = 1;
+					// It might be a deleted commit or something else.
+					core.error( err as Error );
```

Comment:
Same thing here about `err` maybe not being `Error`. Perhaps in both cases just use `as!`.